### PR TITLE
bug/isal

### PR DESCRIPTION
**Pull request by willkoh**
This PR contains the following changes:
- Correctly uses the ISA-L that is from the open source repo provided by Intel. Previously, we were using a non-official version. Works the same way, so just had to change the link and the way we built the library.
- Uses a build of hadoop3 that was compiled from source that uses the ISA-L functionalities. Previously, rdfs could use ISA-L but the hadoop clients could not because it wasn't properly linked. With this new compiled version, both parties can now use ISA-L to ensure fast EC.

### CODE REVIEW

File: .ci/install/isal.sh [hunk 1]
```diff
@@ -2,15 +2,22 @@
...
 
-ISAL_MIRROR=http://kevinlin.web.rice.edu/static/isal.tar.gz
+ISAL_MIRROR=http://kevinlin.web.rice.edu/static/isal-2.tar.gz
 
 cd $HOME
-if [ ! -d /tmp/cache/isal ]; then
-    wget --quiet -O isal.tar.gz $ISAL_MIRROR
-    tar -xf isal.tar.gz -C /tmp
-    mv /tmp/isa-l_open_src_2.13 /tmp/cache/isal
+if [ -d /tmp/cache/isal ]; then
```
Comment (LINKIWI) [inline .ci/install/isal.sh]: This step isn't necessary--we can clear the build cache from Travis directly. If you don't want ISAL to go through the build cache you can just remove this entire `if` block and go here: https://travis-ci.org/comp413-2017/RDFS/caches
and click on "Delete all repository caches"

File: .ci/install/isal.sh [hunk 2]
```diff
@@ -2,15 +2,22 @@
...
     cd /tmp/cache/isal
-    make
+    make clean
     cd $HOME
+    rm -rf /tmp/cache/isal
 fi
+wget --quiet -O isal.tar.gz $ISAL_MIRROR
+tar -xf isal.tar.gz -C /tmp
+mv /tmp/isal /tmp/cache/isal
```
Comment (LINKIWI) [inline .ci/install/isal.sh]: Following up to the above comment: you should probably just extract the tarball into $HOME, cd and build in that directory, and remove the last line of the script that symlinks it to `/tmp/cache/isal`.

**Review by LINKIWI — APPROVED** [review]
Looks good after addressing comments

**Comment by willkoh** [comment]
@LINKIWI Made those changes! Will merge after these checks pass!

File: .ci/install/isal.sh [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
-ln -s /tmp/cache/isal isal
+wget --quiet -O isal.tar.gz $ISAL_MIRROR
+tar -xf isal.tar.gz
+cd $HOME/isal
+./autogen.sh
+./configure
+make
+sudo make install
+cd $HOME
+ln -s $HOME/isal isal
```
Comment (LINKIWI) [inline .ci/install/isal.sh:16]: Does this work? Seems like it's creating a symlink to itself? I think you can probably delete this line entirely

Comment (willkoh) [reply .ci/install/isal.sh:16]: shoot accidentally merged without seeing this. will make a quick oneliner pr for this. my bad.

**Pull request merged**